src/commands/push.js

```diff
@@ -1,36 +1,70 @@
 import fs from 'fs-extra';
+import path from 'path';
+import { prompt } from 'inquirer';
+import semver from 'semver';
+import _ from 'lodash';
+
 import * as extensionManager from '../clients/extension-manager';
-import {getHostEnvName} from '../clients/server-env';
 import * as local from '../clients/local-extensions';
+import { getHostEnvName } from '../clients/server-env';
 import {
   ensureInExtensionDir,
   getExtensionCanonicalName,
   loadExtensionJson,
-  saveExtensionJson
+  saveExtensionJson,
 } from '../services/extension';
+import extLint from '../services/extlint';
 import shoutemPack from '../services/packer';
+import { createProgressHandler } from '../services/progress-bar';
+import { spinify, startSpinner } from '../services/spinner';
+import { canPublish } from '../clients/extension-manager';
 import msg from '../user_messages';
-import _ from 'lodash';
-import {createProgressHandler} from '../services/progress-bar';
-import {spinify, startSpinner} from '../services/spinner';
-import extLint from '../services/extlint';
-import {ensureUserIsLoggedIn} from "./login";
-import {canPublish} from "../clients/extension-manager";
-import { prompt } from 'inquirer';
-import semver from 'semver';
+import { ensureUserIsLoggedIn } from './login';
+
+function ensureRequiredFilesExist(dir, extTitle) {
+  const requiredFiles = [
+    'app/package.json',
+    'app/index.js',
+    'server/package.json',
+  ];
+
+  process.stdout.write('Checking for required files...');
+  _.forEach(requiredFiles, (fileName) => {
+    if (!fs.existsSync(path.join(dir, fileName))) {
+      console.log('\n');
+      throw new Error(msg.push.missingRequiredFile(fileName, extTitle));
```

Comment:
Let's throw error after we've checked for all the files, and given the full report ? So you don't have to run the script for every missing file